src/v2/view-builder/views/ov/ChallengeAuthenticatorOktaVerifyView.js

```diff
@@ -1,23 +1,23 @@
 import BaseAuthenticatorView from '../../components/BaseAuthenticatorView';
 import { 
-  Body as ChallengeAuthenticatorOktaVerifyPushOnlyView
+  Body as ChallengeAuthenticatorDataOktaVerifyPushOnlyView
 } from './ChallengeAuthenticatorDataOktaVerifyPushOnlyView';
-import { Body as ChallengeAuthenticatorOktaVerifyView } from './ChallengeAuthenticatorDataOktaVerifyView';
+import { Body as ChallengeAuthenticatorDataOktaVerifyView } from './ChallengeAuthenticatorDataOktaVerifyView';
 
 export default BaseAuthenticatorView.extend({
   initialize() {
     BaseAuthenticatorView.prototype.initialize.apply(this, arguments);
     if (this.isPushOnlyFlow()) {
-      this.Body = ChallengeAuthenticatorOktaVerifyPushOnlyView;
+      this.Body = ChallengeAuthenticatorDataOktaVerifyPushOnlyView;
     } else {
-      this.Body = ChallengeAuthenticatorOktaVerifyView;
+      this.Body = ChallengeAuthenticatorDataOktaVerifyView;
     }
   },
 
   isPushOnlyFlow() {
-    const uiSchema = this.options.currentViewState.uiSchema;
+    const uiSchema = this.options?.currentViewState?.uiSchema;
```

Comment:
were you not able to use `authenticatorMethodType` of AppState to get the methodType ?

Reply:
OR this.model.get('authenticator.methodType') ??

Reply:
Using `this.options.appState.get('authenticatorMethodType')`  returns the value `push` but here its required to get the possible options under the method type 'push'. this is how the uiSchema object looks and the goal is to find out whether there is just one option and its of value `push`. Options could have multiple values. based on what it has, we are deciding on the view. Said that, do you think there is a better way to achieve the same?

```{
    "name": "authenticator.methodType",
    "required": true,
    "options": [
        {
            "value": "push",
            "label": "Get a push notification"
        },
        {
            "value": "totp",
            "label": "Enter a code"
        },
        {
            "value": "signed_nonce",
            "label": "Use Okta FastPass"
        }
    ],
    "value": "push",
    "label-top": true,
    "data-se": "o-form-fieldset-authenticator.methodType",
    "type": "radio"
}
```

Reply:
See if this method is helpful to count the authenticator options ? `getRemediationAuthenticationOptions`

Reply:
Discussed offline. `getRemediationAuthenticationOptions` method does not help here as the api response structure is different. Would be using `getSchemaByName` instead and from that can get the `options`.